### PR TITLE
Include the room ID in more purge-room log lines.

### PR DESCRIPTION
**Pull request by clokep**
Otherwise if you have multiple purges at once then you can't tell which is progressing.

### CODE REVIEW

**Comment by reivilibre** [comment]
Not necessary, but in case you didn't think about it and would have liked it if you did: How about putting the room ID in the log context? We do this somewhere for some things (partial state resync?)

**Comment by clokep** [comment]
> Not necessary, but in case you didn't think about it and would have liked it if you did: How about putting the room ID in the log context? We do this somewhere for some things (partial state resync?)

Do you have an example? How does this get formatted for log lines? (I think it'll show up in structured logs...)

**Comment by reivilibre** [comment]
> > Not necessary, but in case you didn't think about it and would have liked it if you did: How about putting the room ID in the log context? We do this somewhere for some things (partial state resync?)
> 
> (I didn't think of it!) Do you have an example? How does this get formatted for log lines? (I think it'll show up in structured logs...)

An example seems to be: https://github.com/matrix-org/synapse/blob/6cddf24e361fe43f086307c833cd814dc03363b6/synapse/federation/federation_server.py#L1238

An example log context is `_process_incoming_pdus_in_room_inner-41-$someeventid`.

I guess it's for when you process multiple things and want to split your log context for each separate thing, but I'm not sure it'd be the worst thing in the world to have a nested log context with the room ID.

**Comment by clokep** [comment]
Ah-ha, I must have forgotten about `nested_logging_context`. It sounds like it would probably be useful?

**Comment by clokep** [comment]
I've switched to the nested logging contexts. 👍

File: synapse/storage/databases/main/purge_events.py [hunk 1]
```diff
@@ -325,20 +325,23 @@ async def purge_room(self, room_id: str) -> List[int]:
         # We then run the same purge a second time without this isolation level to
         # purge any of those rows which were added during the first.
 
+        logger.info("[purge] Starting initial main purge of [1/2]]")
```
Comment (squahtx) [inline synapse/storage/databases/main/purge_events.py:328]:
```suggestion
        logger.info("[purge] Starting initial main purge of [1/2]")
```

File: synapse/storage/databases/main/purge_events.py [hunk 2]
```diff
@@ -325,20 +325,23 @@ async def purge_room(self, room_id: str) -> List[int]:
...
         state_groups_to_delete = await self.db_pool.runInteraction(
             "purge_room",
             self._purge_room_txn,
             room_id=room_id,
             isolation_level=IsolationLevel.READ_COMMITTED,
         )
 
+        logger.info("[purge] Starting secondary main purge of [2/2]]")
```
Comment (squahtx) [inline synapse/storage/databases/main/purge_events.py:336]: think we've got a stray bracket here
```suggestion
        logger.info("[purge] Starting secondary main purge of [2/2]")
```

**Review by squahtx — APPROVED** [review]
Looks good!

**Pull request merged**